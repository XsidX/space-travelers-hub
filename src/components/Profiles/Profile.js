import React from 'react';
import propTypes from 'prop-types';

export const Profile = ({ children, title }) => {
  return (
    <ul className="profile">
      <li>
        <h2 className="profile__title">{title}</h2>
      </li>
      {children}
    </ul>
  );
};

export const ProfileDetail = ({ name }) => {
  return (
    <li className="profile__content-wrapper">
      <h3 className="profile__content-name">{name}</h3>
    </li>
  );
};

Profile.propTypes = {
  children: propTypes.node.isRequired,
  title: propTypes.string.isRequired,
};

ProfileDetail.propTypes = {
  name: propTypes.string,
};

ProfileDetail.defaultProps = {
  name: '',
};
