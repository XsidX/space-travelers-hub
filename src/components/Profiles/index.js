import React from 'react';
import { useSelector } from 'react-redux';
import { Profile, ProfileDetail } from './Profile';

const Profiles = () => {
  const missions = useSelector((state) => state.missions.missions);
  const joinedMissions = missions.filter((mission) => mission.joined);

  return (
    <div className="profile-grid">
      <Profile title="My Missions">
        {joinedMissions.length > 0 &&
          joinedMissions.map(({ id, name }) => (
            <ProfileDetail key={id} name={name} />
          ))}
        {joinedMissions <= 0 && <ProfileDetail name="No Missions Joined" />}
      </Profile>
    </div>
  );
};

export default Profiles;
