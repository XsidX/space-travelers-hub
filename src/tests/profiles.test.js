import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';
import Profiles from '../components/Profiles/index';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Profiles component', () => {
  const { useSelector } = reactRedux;

  beforeEach(() => {
    useSelector.mockImplementation((selector) => selector({
      missions: {
        missions: [],
      },
      rockets: {
        rockets: [],
      },
    }));
  });

  afterEach(() => {
    useSelector.mockClear();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Profiles />);

    expect(tree).toMatchSnapshot();
  });
});
