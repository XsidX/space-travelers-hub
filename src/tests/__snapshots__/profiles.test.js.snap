// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profiles component renders correctly 1`] = `
<div
  className="profile-grid"
>
  <ul
    className="profile"
  >
    <li>
      <h2
        className="profile__title"
      >
        My Missions
      </h2>
    </li>
    <li
      className="profile__content-wrapper"
    >
      <h3
        className="profile__content-name"
      >
        No Missions Joined
      </h3>
    </li>
  </ul>
  <ul
    className="profile"
  >
    <li>
      <h2
        className="profile__title"
      >
        My Rockets
      </h2>
    </li>
    <li
      className="profile__content-wrapper"
    >
      <h3
        className="profile__content-name"
      >
        No Rocket Reserved
      </h3>
    </li>
  </ul>
</div>
`;
